api/v1beta1/types.go

```diff
@@ -123,6 +161,15 @@ type LoadBalancerSpec struct {
 	// +kubebuilder:validation:Pattern=`(^[1-9][0-9]{0,31}$)|(^[a-z][a-z0-9-]{4,28}[a-z0-9]$)`
 	// +optional
 	APIServerInstanceGroupTagOverride *string `json:"apiServerInstanceGroupTagOverride,omitempty"`
+
+	// LoadBalancerType defines the type of Load Balancer that should be created.
+	// If not set, a Global External Proxy Load Balancer will be created by default.
+	// +optional
+	LoadBalancerType *LoadBalancerType `json:"loadBalancerType,omitempty"`
```

Comment:
I think you can achieve those via kubebuilder validations :
 "// +kubebuilder:validation:UniqueItems=" to make sure array entries are unique
 and "// +kubebuilder:default:="External" will make sure it would not be empty.
 
 I know I had suggested making LoadBalancers of type []LoadBalancerType, with the array containing `ExternalLoadBalancerOnly` and/or `InternalLoadBalancerOnly` but looking at the rest of the implementation, I am not sure that is the right path.

Reply:
Are the 4 optional fields added to Network type allowed to be empty when LoadBalancerType is InternalLoadBalancerOnly/DualLoadBalancer ?

Reply:
Also, can the value of LoadBalancerType change once set i.e is it immutable? If so, do we need a validateUpdate() webhook preventing an update for LoadBalancerType?

Reply:
> Are the 4 optional fields added to Network type allowed to be empty when LoadBalancerType is InternalLoadBalancerOnly/DualLoadBalancer ?

Yes, these marked as `+optional` and are string pointers so its OK to not be set

Reply:
> Also, can the value of LoadBalancerType change once set i.e is it immutable? If so, do we need a validateUpdate() webhook preventing an update for LoadBalancerType?

That's a good point, thanks for catching that. I will add a webhook check for it